src/server/scripts/Kalimdor/HallsOfOrigination/boss_anraphet.cpp

```diff
@@ -327,24 +336,37 @@ class npc_brann_bronzebeard_anraphet : public CreatureScript
     public:
         npc_brann_bronzebeard_anraphet() : CreatureScript("npc_brann_bronzebeard_anraphet") { }
 
+        bool OnGossipHello(Player* player, Creature* creature) override
+        {
+            InstanceScript* instance = creature->GetInstanceScript();
+            if (!instance || instance->GetBossState(DATA_VAULT_OF_LIGHTS) == DONE)
+                return false;
+
+            player->PrepareGossipMenu(creature, GOSSIP_MENU_WE_ARE_READY, true);
+            player->SendPreparedGossip(creature);
+            return true;
```

Comment:
this is doable with conditions

Reply:
I see. Conditions are already there... https://pastebin.com/nWdeR2fL
But creature_template only has slot for one menu id, how does it find others?

Reply:
`creature_template` only lists the default `gossip_,menu_id` for a creature. To access gossip menus or gossip menu options, link them via the `gossip_menu` table and use conditions for when they are shown to the player.

Reply:
if you need a gossip menu option to open a new gossip menu, it is done via the `gossip_menu_option`.`action_menu_id` field.

Reply:
npc must show another ```gossip_menu``` on hello, and there is no link between second menu and ```creature_template``` :/ i guess i will just change ```creature_template```.```gossip_menu_id``` manually

Reply:
Which creature ID (`creature_template`.`entry`) and `creature_template`.`gossip_menu_id` are the original ones in this case?

Reply:
Brann 39908
Menu that starts the encounter 11339

Reply:
Well, it can possibly be set in `conditions` by using `CONDITION_SOURCE_TYPE_GOSSIP_MENU` 14, `SourceGroup` (gossip menu entry) 11339, SourceEntry (gossip_menu.text_id) 15794 and with `ConditionTypeOrReference` `CONDITION_INSTANCE_INFO` 13 -- see https://trinitycore.atlassian.net/wiki/spaces/tc/pages/2130002/conditions#conditions-ConditionTypeOrReference for more info. Maybe @Killyana can suggest some more to help you along the way here.

Reply:
[198_misc.sql#L2](https://github.com/TrinityCore/TrinityCore/blob/a249d86b00eb0854cb925f8bea76ddbb364208df/sql/old/4.3.4/TDB1_to_TDB2_updates/world/198_misc.sql#L2) - here, back in 2014, it was set so it works, but sniffs overwrite values.
Anyway, for now it seems only C++ can handle menu change...

Reply:
That's odd. I wonder why the `gossip_menu_id` changed, but not the `conditions` table or `gossip_menu` table content for this encounter.

Reply:
He ends up with different gossip text after you kill Anraphet... different ```gossip_menu_id ```. Probably that's why